src/frontend/index.js

```diff
@@ -28,24 +29,52 @@ logger.log(logPrefix + 'Loaded at ' + (new Date()));
 function init() {
 	var dispatcher = new EventEmitter();
 	
+	var config = null;
+	var dataHostnamesRoundRobin = 0;
+	
 	var backendApi = {
-		baseUrl: config.backend.httpBaseUrl,
-		normalizeUrl: function (url) {
-			if (!/^(https?:)?\/\//.test(url)) {
-				url = this.baseUrl + url;
-			}
-			return url;
+		/**
+		 * Loads the config from the backend, then loads the layout.
+		 * The config includes "layoutUrl" which is the base for other URLs.
+		 * The config also includes "dataHostnames" an array of hostnames 
+		 * that resolve to the backend to fool the browser's domain connection limit.
+		 */
+		loadConfig: function () {
+			// Load the config from the backend:
+			var configUrl = '/config.json';
+			
+			$.ajax({
+				url: configUrl,
+				dataType: 'json'
+			}).then(function (response) {
+				config = _.extend({
+					layoutUrl: '/layout',
+					dataHostnames: []
+				}, response);
+				
+				backendApi.loadLayout();
+			}, function (jqXHR) {
+				logger.error(logPrefix + 'Failed to load config from ' + configUrl + ':', jqXHR);
+				
+				global.alert('An error occurred while loading config from ' + configUrl + '.');
+			});
 		},
-		loadLayout: function (layoutUrl) {
+		
+		/**
+		 * Loads the layout via the URL from the config.
+		 */
+		loadLayout: function () {
+			var layoutUrl = config.layoutUrl;
+			
 			$.ajax({
-				url: this.normalizeUrl(layoutUrl),
+				url: layoutUrl,
 				dataType: 'json'
 			}).then(function (response) {
 				var layout = response.layout;
-			
+				
```

Comment:
Answered above. We could agree on empty lines being totally empty if you prefer so.